src/OpenTelemetry/Metrics/ExplicitBucketHistogramConfiguration.cs

```diff
@@ -14,19 +14,78 @@
 // limitations under the License.
 // </copyright>
 
+using System;
+
 namespace OpenTelemetry.Metrics
 {
+    /// <summary>
+    /// Stores configuration for a histogram metric stream with explicit bucket boundaries.
+    /// </summary>
     public class ExplicitBucketHistogramConfiguration : MetricStreamConfiguration
     {
         /// <summary>
-        /// Gets or sets the values representing explicit histogram bucket
-        /// boundary values.
+        /// Gets or sets the optional boundaries of the histogram metric stream.
         /// </summary>
         /// <remarks>
-        /// The array must be in ascending order with distinct values.
-        /// An empty array would result in no histogram buckets being calculated.
-        /// A null value would result in default bucket boundaries being used.
+        /// Requirements:
+        /// <list type="bullet">
+        /// <item>The array must be in ascending order with distinct
+        /// values.</item>
+        /// <item>An empty array would result in no histogram buckets being
+        /// calculated.</item>
+        /// <item>A null value would result in default bucket boundaries being
+        /// used.</item>
+        /// </list>
+        /// Note: A copy is made of the provided array. Boundaries cannot be
+        /// modified after being set.
         /// </remarks>
-        public double[] Boundaries { get; set; }
+        public double[] Boundaries
+        {
+            get
+            {
+                if (this.CopiedBoundaries != null)
+                {
+                    double[] copy = new double[this.CopiedBoundaries.Length];
+                    this.CopiedBoundaries.AsSpan().CopyTo(copy);
+                    return copy;
+                }
+
+                return null;
+            }
+
+            set
+            {
+                if (value != null)
+                {
+                    if (!IsSortedAndDistinct(value))
+                    {
+                        throw new ArgumentException($"Histogram boundaries are invalid. Histogram boundaries must be in ascending order with distinct values.", nameof(value));
+                    }
+
+                    double[] copy = new double[value.Length];
+                    value.AsSpan().CopyTo(copy);
+                    this.CopiedBoundaries = copy;
+                }
+                else
+                {
+                    this.CopiedBoundaries = null;
+                }
+            }
+        }
+
+        internal double[] CopiedBoundaries { get; private set; }
+
+        private static bool IsSortedAndDistinct(double[] values)
+        {
+            for (int i = 1; i < values.Length; i++)
+            {
+                if (values[i] <= values[i - 1])
```

Comment:
Do we check that the buckets boundaries don't contain the value `double.NaN`?
I believe that should also be considered invalid: https://github.com/open-telemetry/opentelemetry-dotnet/pull/2592/files#diff-af5f4b9fc074ecc8cc6df71f33bfa667493b46942717092dfdd7bc2722968bf4R51

One way to do it https://github.com/open-telemetry/opentelemetry-dotnet/pull/2592/files#diff-7464baae7413dc78c7c80dd84bf4d24a656a9ca70aaf432dee74408cfebb9057R255 or we could add something in the setter here https://github.com/open-telemetry/opentelemetry-dotnet/pull/3117/files#diff-a058dbf2817506d9e5cd50a56526bd65f3282af09018d99686c426432816be4cR60 like `if (!IsSortedAndDistinct(value) || value.Any(double.IsNaN))`